idseq_pipeline/commands/common.py

```diff
@@ -281,57 +280,38 @@ def wrapped_operation(*args, **kwargs):
     return wrapped_operation
 
 
-def execute_command_with_output(command, progress_file=None, timeout=None, grace_period=None):
+def execute_command(command, progress_file=None, timeout=None, grace_period=None, with_output=False):
     with CommandTracker() as ct:
         with print_lock:
             print "Command {}: {}".format(ct.id, command)
         with ProgressFile(progress_file):
-            # Capture only stdout.   Send stderr to parent stderr.   Take input from parent stdin.
-            ct.proc = subprocess.Popen(command, shell=True, stdin=sys.stdin.fileno(), stdout=subprocess.PIPE, stderr=sys.stderr.fileno())
             if timeout:
                 ct.timeout = timeout
             if grace_period:
                 ct.grace_period = grace_period
-            stdout, _undefined = ct.proc.communicate()
+            if with_output:
+                # Capture only stdout.  Child stderr = parent stderr.   Child input = parent stdin.
+                ct.proc = subprocess.Popen(command, shell=True, stdin=sys.stdin.fileno(), stdout=subprocess.PIPE, stderr=sys.stderr.fileno())
+                stdout, _ = ct.proc.communicate()
+            else:
+                # Capture nothing.  Child inherits parent stdin/out/err.
+                ct.proc = subprocess.Popen(command, shell=True)
+                ct.proc.wait()
+                stdout = None
             if ct.proc.returncode:
                 raise subprocess.CalledProcessError(ct.proc.returncode, command, stdout)
-            return stdout
+            if with_output:
+                return stdout
 
 
-def execute_command_realtime_stdout(command, progress_file=None):
-    with CommandTracker() as ct:
-        with print_lock:
-            print "Command {}: {}".format(ct.id, command)
-        with ProgressFile(progress_file):
-            subprocess.check_call(command, shell=True)
-
-
-def execute_command(command, progress_file=None):
-    execute_command_realtime_stdout(command, progress_file)
+def execute_command_with_output(command, progress_file=None, timeout=None, grace_period=None):
+    return execute_command(command, progress_file, timeout, grace_period, with_output=True)
 
 
 def remote_command(base_command, key_path, remote_username, instance_ip):
     return 'ssh -o "StrictHostKeyChecking no" -o "ConnectTimeout 15" -i %s %s@%s "%s"' % (key_path, remote_username, instance_ip, base_command)
 
 
-def check_s3_file_presence(s3_path):
```

Comment:
As the comment here suggests, this test is totally unreliable.   Doing it right with the s3 head-object operation requires careful testing of error codes, and still seems brittle.   The bottom line is that you can never be certain a file doesn't exist in s3, you can only increase your belief by retrying, which has a cost.   So it's best to avoid using this function --- and luckily, we don't actually need it.